### PR TITLE
Dipto task/ecs fargate task - video conversion with ecs task throug docker image.

### PR DESCRIPTION
**Pull request by D-sourav-M044**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [X] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Video conversion is now performing with ecs task.
## What was done
New ecs task has been created with serverless. Required roles and policy has been added with the task. Through this task ecs run a docker image that converted a video that has been uploaded to s3 bucket. After successfully converted the video, the updated/scaled/converted video is again uploaded to the same s3 bucket and the name is put into dynamo table and so on.
## How to test
Need to upload a video file into the s3 bucket. After waiting some second the updated video will be uploaded into same s3 bucket and updated name will be uploaded to the dynamo table.
- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -204,6 +223,70 @@ resources:
...
+              Version: "2012-10-17"
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - "ecr:GetAuthorizationToken"
+                    - "ecr:BatchGetImage"
+                    - "ecr:BatchCheckLayerAvailability"
+                    - "ecr:GetDownloadUrlForLayer"
+                    - "ecs:*"
+                    - "logs:*"
```
Comment (mhazad) [inline serverless.yml:256]: Pls limit the permission

File: source/handler.js [hunk 2]
```diff
@@ -335,8 +338,75 @@ const videoConversion = (fileName) => {
...
+    return {
+      status: true,
+      statusCode: 200,
+      message: 'Video converted successfully'
+    };
+  } catch (err) {
+    console.log(err.message);
+    return {
+      status: false,
+      statusCode: 400,
```
Comment (mhazad) [inline source/handler.js:403]: This should not be 400 error. Please fix it.

**Review by mhazad — CHANGES_REQUESTED** [review]
Please address the feedback.

File: source/handler.js (hunk 2 above)
Comment (D-sourav-M044) [reply source/handler.js]: updated the statusCode. [ba5ba8f](https://github.com/CraftsmenLtd/coe-batch-q322/commit/ba5ba8fe85506261a2d7d617b2f1eb79e486751a)

File: serverless.yml (hunk 1 above)
Comment (D-sourav-M044) [reply serverless.yml]: Here "ecs:*"  is unnecesssary- as the task will use ECR to use the docker image. And ECS task creation with required configuration has been done in yml file. So I have removed this  - "ecs:*". Also limited the log permission defined as in documentation. After changing the yml everything works good. [ba5ba8f](https://github.com/CraftsmenLtd/coe-batch-q322/commit/ba5ba8fe85506261a2d7d617b2f1eb79e486751a)

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**